### ISSUE TITLE
Units of measure in numbers don't work as documented.

### ISSUE THREAD

**Issue by arianvp** [hashicorp/terraform#3287]
Version of terraform: v0.6.3

The [documentation](https://www.terraform.io/docs/configuration/syntax.html) says that numbers should support units of measure. Meaning `1gb` should be a valid number. Except currently that doesn't seem to parse.

> * Numbers are assumed to be base 10. If you prefix a number with 0x, it is treated as a hexadecimal number.
> * Numbers can be suffixed with kKmMgG for some multiple of 10. For example: 1k is equal to 1000.
> * Numbers can be suffixed with [kKmMgG]b for power of 2 multiples, example: 1kb is equal to 1024.

Say we have:

```hcl
variable "size" {
	default = 10gb
}

output "size_in_bytes" {
	value = "${var.size}"
}
```

it should work but instead we get:

```
Error loading config: Error parsing example.tf: Line 4, column 1: syntax error
```

I also tried `default = "10gb"` (enclosed in double quotes) but then we get:

```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

  size_in_bytes = 10gb
```

i.e. the size is parsed as string.

**Comment by arianvp** [comment]
If you guys can give me some pointers where to look, I can give a shot at trying to tackle this.

**Comment by jjshoe** [comment]
https://terraform.io/docs/providers/aws/r/ebs_volume.html - "size - (Optional) The size of the drive in GB."

**Comment by arianvp** [comment]
I know.  This was just a random example of a "valid" configuration file.  The bug is not with the amazon provider, but with the fact that the documentation says units of measure are supported in number literals while they aren't

**Comment by radeksimko** [comment]
Per discussion on IRC, I modified your original example, so it's more obvious that this problem is provider/resource unrelated, yet still valid.

**Comment by radeksimko** [comment]
We also discussed on IRC, that it would be useful to have a way to convert bytes to other units, so that one variable can be useful for multiple resources, each expecting different units.

```hcl
variable "size" {
  default = "1TB"
}

resource "aws_ebs_volume" "example" {
  size = "${in_gb(var.size)}"
}
```

https://botbot.me/freenode/terraform-tool/2015-09-21/?msg=50178901&page=1

> @arianvp that's actually pretty useful I think. For example scaleway defines their volumes in bytes. But you could then use the same variable for defining volumes sizes on their cloud and amazon's cloud.

**Comment by maxbeatty** [comment]
I just came up against this when trying to create an elasticsearch domain on AWS and setting the [`volume_size` for ebs](https://terraform.io/docs/providers/aws/r/elasticsearch_domain.html#volume_size)

```
resource "aws_elasticsearch_domain" "example" {
  domain_name = "example"

  ebs_options {
    ebs_enabled = true
    volume_type = "gp2" # General Purpose
    volume_size = 10Gb
  }

  snapshot_options {
    automated_snapshot_start_hour = 23
  }
}
```

```
$ terraform plan
Error loading config: Error parsing /path/to/es.tf: Line 12, column 3: syntax error
```

It would be great if "10Gb" expanded to the proper integer for the AWS API call

**Comment by dmacvicar** [comment]
With Terraform 0.6.12 I get no syntax error, but it looks like the suffixes do not have anny effect

```hcl
variable "size" {
    default = 10kb
}

output "size_in_bytes" {
    value = "${var.size}"
}
```

```console
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

  size_in_bytes = 10
```

**Comment by dennybaa** [comment]
+1

**Comment by mitchellh** [comment]
@dmacvicar Verified that that will give a syntax error now.

So this is a huge misdocumentation. We had this feature as part of Terraform 0.1 and 0.2 but it has probably been broken since then since when we switched to HCL as the format, this didn't come along with it. We've chatted about it and decided to not support this feature (which has never really worked in any meaningful moment of history). 

Closing this and will update docs. Sorry!
